### PR TITLE
Defining types for Hash and HighQcRoundSigTuple

### PR DESCRIPTION
**Pull request by michael-yxchen**
Based on #81 
- Defining new types for them so the conversion function can be defined in the crate defining the rust types
- This is an intermediate change to keep PR smaller. It breaks the build so must be merged together with #84

### CODE REVIEW

File: monad-types/src/lib.rs [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl AsRef<[u8]> for Hash {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
```
Comment (michael-yxchen) [inline monad-types/src/lib.rs:24]: `Hash` type

File: monad-consensus/src/types/timeout.rs [hunk 2]
```diff
@@ -23,17 +23,29 @@ impl Hashable for HighQcRound {
...
+    fn as_ref(&self) -> &[u8] {
+        self.qc_round.as_ref()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct HighQcRoundSigTuple<S> {
+    pub high_qc_round: HighQcRound,
+    pub author_signature: S,
+}
```
Comment (michael-yxchen) [inline monad-consensus/src/types/timeout.rs:36]: `HighQcRoundSigTuple` type

File: monad-consensus/src/types/timeout.rs [hunk 3]
```diff
@@ -23,17 +23,29 @@ impl Hashable for HighQcRound {
     }
 }
 
+impl AsRef<[u8]> for HighQcRound {
+    fn as_ref(&self) -> &[u8] {
+        self.qc_round.as_ref()
+    }
+}
```
Comment (michael-yxchen) [inline monad-consensus/src/types/timeout.rs:30]: so Hasher.update() can take in `HighQcRound`

**Comment by omegablitz** [comment]
Should we just squash this into #84 if it's needed for the tests to pass? It'd be nice if every commit in master passes the tests in case we want to git bisect or something in the future.

**Comment by michael-yxchen** [comment]
Sure, I'll highlight the new types squashing it into #84.

**Pull request closed without merging**